### DISCUSSION TITLE
[Feature] Workers to handle the load

### DISCUSSION THREAD

**Discussion by 1mfaasj** [immich-app/immich#5189] [Feature Request]
### The feature

I am familiar with the (hardware) requirements (https://immich.app/docs/install/requirements)
However, I am missing a way to distribute load. (with workers for example). It would be nice if we could specify Workers=1 in the docker compose environment (variables), so that this is taken into account when executing the jobs (e.g. to create thumbnails). For example, if you enter Workers=2, the jobs will take up more capacity. This also allows smaller servers to use Immich.

What I now notice is that I have a lot of photos and Immich is barely making any progress on a raspberry. keeps getting stuck in a timeout.

the "competitor" Photoprism does this. see here: https://docs.photoprism.app/getting-started/config-options/
that works very well for me.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
You can run multiple instances of the microservices container, jobs will automatically be distributed between them.
